examples/routing_example/common/index.ts

```diff
@@ -20,5 +20,8 @@ export const DEPRECATED_ROUTES = {
   DEPRECATED_ROUTE: '/api/routing_example/d/deprecated_route',
   REMOVED_ROUTE: '/api/routing_example/d/removed_route',
   MIGRATED_ROUTE: '/api/routing_example/d/migrated_route',
-  VERSIONED_ROUTE: '/api/routing_example/d/versioned',
+  VERSIONED_ROUTE: '/api/routing_example/d/versioned_route',
+  INTERNAL_DEPRECATED_ROUTE: '/api/routing_example/d/internal_deprecated_route',
```

Comment:
These are great for testing within Kibana, where the internal origin header is added to a request sent from console. 

However, we need to surface calls to internal APIs **mainly** when requests originate externally and may not necessarily include the header. Surfacing these will warn end-users about the upcoming change.

We also want telemetry, to track adoption of internal APIs that are being consumed by integrations and use this info to consider public API alternatives instead,

In other words:
```
server.restrictInternalApis = true // commenting out also works
```
desired: 
- surface external requests to internal APIs only (current behavior of this PR)
- increment counters for routes called externally from Kibana (ATM, usage counters include other routes such as `unversioned|get|/XXXXXXXXXXXX/bundles/plugin/transform/1.0.0/{path*}`, `unversioned|get|/XXXXXXXXXXXX/bundles/plugin/discover/1.0.0/{path*}` and `unversioned|get|/XXXXXXXXXXXX/bundles/plugin/aiops/1.0.0/{path*}`))
- no counters for routes called externally 😞 

scenario 2: 
```
server.restrictInternalApis = false //explicitly opt-in to use internal APIs.
```
desired: 
- surface external requests to internal APIs only (only when requests don't include the origin header)
- log warning when restriction's disabledt
- usage counters: still logging `****/gundle/**`

Reply:
I dont fully understand the comment here. correct `bundles` are `public` apis so they should not be tracked. 

I've updated the README and console commands to explicitly set `elasticInternalOrigin=false` to enable tracking them from the console for quick tesitng. I did add integration tests to ensure proper testing for the behavior.

We need to explicitly set the query param `elasticInternalOrigin` to `false` to track the request as non-internal origin.

```
# Route deprecations for Versioned routes: Version 1 is deprecated
GET kbn:/api/routing_example/d/versioned_route?apiVersion=2023-10-31&elasticInternalOrigin=false

# Route deprecations for Non-versioned routes?elasticInternalOrigin=false
GET kbn:/api/routing_example/d/removed_route?elasticInternalOrigin=false
GET kbn:/api/routing_example/d/deprecated_route?elasticInternalOrigin=false
POST kbn:/api/routing_example/d/migrated_route?elasticInternalOrigin=false
{}

# Access deprecations
GET kbn:/api/routing_example/d/internal_deprecated_route?elasticInternalOrigin=false
GET kbn:/internal/routing_example/d/internal_only_route?elasticInternalOrigin=false
GET kbn:/internal/routing_example/d/internal_versioned_route?apiVersion=1&elasticInternalOrigin=false
```


> These are great for testing within Kibana, where the internal origin header is added to a request sent from console.
> However, we need to surface calls to internal APIs mainly when requests originate externally and may not necessarily include the header. Surfacing these will warn end-users about the upcoming change.

Can you clarify what you mean exactly here? External vs internal calls are specified via the header (or query param), and if the header/queryParam is missing then its an external call which in that case qualifies to be surfaced in the deprecations as access api deprecation (Internal access + the requesting caller is external).

Reply:
Retested and the changes look great! Both scenarios I was testing for work as expected. 
With the restriction explicitly disabled in `kibana.yml`, we get a deprecation entry in UA when the API is called without the internal origin header and the usage counters increment, consistent with logging added in https://github.com/elastic/kibana/pull/195696.